src/core/base_stream_class.js

```diff
@@ -0,0 +1,99 @@
+/* Copyright 2021 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { shadow, unreachable } from "../shared/util.js";
+
+class BaseStreamClass {
```

Comment:
I think it would be better to remove `Class` from the class name and the file name. It's already visible that it's a class and we don't do this elsewhere (compare e.g., `BasePreferences`, `BaseLocalCache`, `BaseCanvasFactory` and more).

Reply:
I was going back and forth quite on lot on the naming here, since I was a little bit worried about `BaseStream` sounding just like any other `...Stream`. However, given that that one cannot create an instance of it, but only extend it, I suppose that worry is unfounded; I'll change the name as suggested :-)